Domains/Jobs/WorkerConfiguration.php

```diff
@@ -0,0 +1,338 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * @file Domains/Jobs/Job.php
+ *
+ * Copyright (c) 2014-2022 Simon Fraser University
+ * Copyright (c) 2000-2022 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class Job
+ * @ingroup domains
+ *
+ * @brief Laravel Eloquent model for Jobs table
+ */
+
+namespace PKP\Domains\Jobs;
+
+class WorkerConfiguration
+{
+    /**
+     * The name of the worker.
+     *
+     * @var string
+     */
+    protected $name = 'default';
+
+    /**
+     * Calculate the number of seconds to wait before retrying the job.
+     *
+     * @var int
+     */
+    protected $backoff = 0;
+
+    /**
+     * Job's worker allowed memory configuration, as the maximum amount of RAM (in megabytes) the worker may consume.
+     *
+     * @var int
+     */
+    protected $memory = 128;
+
+    /**
+     * Job's worker timeout configuration, as the maximum number of seconds a child worker may run.
+     *
+     * @var int
+     */
+    protected $timeout = 30;
+
+    /**
+     * Job's worker sleep configuration, as the number of seconds to wait/sleep in between polling the queue.
+     *
+     * @var int
+     */
+    protected $sleep = 3;
+
+    /**
+     * Number of times to attempt a job before logging it failed
+     *
+     * @var int
+     */
+    protected $tries = 3;
+
+    /**
+     * Job's worker force configuration, indicates if the worker should run even in Laravel's maintenance mode.
+     *
+     * @var bool
+     */
+    protected $force = false;
+
+    /**
+     * Job's worker stopWhenEmpty configuration, indicates if the worker should stop when queue is empty.
+     *
+     * @var bool
+     */
+    protected $stopWhenEmpty = false;
+
+    /**
+     * The number of jobs to process before stopping
+     *
+     * @var int
+     */
+    protected $maxJobs = 0;
+
+
+    /**
+     * The maximum number of seconds the worker should run
+     *
+     * @var int
+     */
+    protected $maxTime = 0;
+
+
+    /**
+     * Number of seconds to rest between jobs
+     *
+     * @var int
+     */
+    protected $rest = 0;
+
+    /**
+     * Set the configurations
+     *
+     * @param array $options
+     * @return self
+     */
+    public static function withOptions(array $options = []) : self
+    {
+        $self = new static;
+
+        collect($options)
+            ->each(fn($value, $option) => method_exists($self, 'set' . ucfirst($option)) 
+                ? $self->{'set' . ucfirst($option)}($value) 
+                : null
```

Comment:
What do you think about throwing an exception when a matching setter function can't be found?

Reply:
good idea . done .